pde_superresolution/weno_test.py

```diff
@@ -28,21 +28,24 @@
 class WENOTest(parameterized.TestCase):
 
   def test_calculate_omega_smooth(self):
-    u = np.zeros(5)
+    u = np.linspace(0, 1)
     actual = weno.calculate_omega(u)
-    expected = np.stack(5 * [[0.1, 0.6, 0.3]], axis=1)
+    actual = actual[:, 2:-2]
+    expected = np.stack(actual.shape[1] * [[0.1, 0.6, 0.3]], axis=1)
     np.testing.assert_allclose(actual, expected)
 
   def test_left_coefficients_smooth(self):
-    u = np.zeros(5)
+    u = np.linspace(0, 1)
     actual = weno.left_coefficients(u)
-    expected = np.stack(5 * [[2/60, -13/60, 47/60, 27/60, -3/60]], axis=1)
+    actual = actual[:, 2:-2]
+    expected = np.stack(actual.shape[1] * [[2 / 60, -13 / 60, 47 / 60, 27 / 60, -3 / 60]], axis=1)
     np.testing.assert_allclose(actual, expected)
 
   def test_right_coefficients_smooth(self):
-    u = np.zeros(5)
+    u = np.linspace(0, 1)
     actual = weno.right_coefficients(u)
-    expected = np.stack(5 * [[-3/60, 27/60, 47/60, -13/60, 2/60]], axis=1)
+    actual = actual[:, 2:-3] # TODO: Note that we need 3 here while in the others 2 is enough
```

Comment:
I think this actually makes sense because we reconstruct the signal (and coefficients) on cell boundaries, i.e., at positions i+1/2.

More explicitly, if `u` was indexed in space at positions `[0, 1, 2, 4, 5, 6, 7, 8, 9, 10]`, then `actual[: 2:-3]` would be indexed at positions `[2.5, 3.5, 4.5, 5.5, 6.5, 7.5]`.

Reply:
Yeah, but then you'd expect `test_left_coefficients_smooth` to work only for `[:, 3:-2]`, no?

Reply:
No, we reconstruct both left and right coefficients at positions i+1/2

Reply:
I'm not sure i follow, but it probably doesn't matter. I think it's ready to merge.